Haro kernel was failing to build on ptolemy due to
```
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn201/dcn201_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/nouveau/nvkm/engine/gr/ctxgk208.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn21/dcn21_hwseq.o
linux-rpi-6.12.47-stable_-aarch64-linux> qemu-aarch64: QEMU internal SIGSEGV {code=MAPERR, addr=0xcb4}
linux-rpi-6.12.47-stable_-aarch64-linux> gcc: internal compiler error: Segmentation fault signal terminated program cc1
linux-rpi-6.12.47-stable_-aarch64-linux> Please submit a full bug report, with preprocessed source (by using -freport-bug).
linux-rpi-6.12.47-stable_-aarch64-linux> See <https://gcc.gnu.org/bugs/> for instructions.
linux-rpi-6.12.47-stable_-aarch64-linux> make[6]: *** [../scripts/Makefile.build:244: drivers/gpu/drm/nouveau/nvkm/engine/gr/ctxgk104.o] Error 4
linux-rpi-6.12.47-stable_-aarch64-linux> make[6]: *** Waiting for unfinished jobs....
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn21/dcn21_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn30/dcn30_hwseq.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn30/dcn30_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn301/dcn301_hwseq.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn301/dcn301_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn302/dcn302_hwseq.o
linux-rpi-6.12.47-stable_-aarch64-linux> make[5]: *** [../scripts/Makefile.build:485: drivers/gpu/drm/nouveau] Error 2
linux-rpi-6.12.47-stable_-aarch64-linux> make[5]: *** Waiting for unfinished jobs....
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn302/dcn302_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn303/dcn303_hwseq.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn303/dcn303_init.o
linux-rpi-6.12.47-stable_-aarch64-linux>   CC [M]  drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn31/dcn31_hwseq.o
```

I have no idea why this is the case, it *seems* to be building fine on arios, but at time of writing is still going at 46+ hours.
Similarly the action is failing due to hitting the 6hr limit.
> Sidenote: according to [this](https://stackoverflow.com/questions/59073731/set-default-timeout-on-github-action-pipeline) it is possible to set a per-step time limit. While this can't be used to increase the 6hr limit it would be good to limit the amount of time the tmate action can idle for.

As for why builds need to happen at all, it turns out the rpi kernels are being moved out of nixpkgs, meaning they are no longer cached:
https://github.com/NixOS/nixos-hardware/pull/1737
https://github.com/NixOS/nixpkgs/pull/477665

--- a/.github/workflows/build_hosts.yml
+++ b/.github/workflows/build_hosts.yml
@@ -21,7 +21,7 @@ jobs:
           primary-key: cache-prepare-${{ runner.os }}-nix-store-${{ hashFiles('**/*.nix') }}
           restore-prefixes-first-match: cache-prepare-${{ runner.os }}-nix-store-
       - name: Export manifest
-        run: nix flake show --json | jq -c -r '.nixosConfigurations | keys' > manifest.json
+        run: nix eval .#githubActions.matrix --json | jq -c -r > manifest.json
       - run: cat manifest.json
       - name: Save manifest to output
         id: manifest
@@ -29,24 +29,25 @@ jobs:
           echo "manifest=$(cat manifest.json)" >> "$GITHUB_OUTPUT"
   build-hosts:
     needs: prepare
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.host.runner }}
     strategy:
       fail-fast: false
       matrix:
         host: ${{ fromJSON(needs.prepare.outputs.manifest) }}
-    env:
-      NIX_CONFIG: extra-platforms = aarch64-linux
+    #env:
+    #  NIX_CONFIG: extra-platforms = aarch64-linux
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
+      #- name: Set up QEMU
+      #  uses: docker/setup-qemu-action@v3
       - uses: wimpysworld/nothing-but-nix@v10
       - uses: DeterminateSystems/nix-installer-action@main
-      - name: Build host ${{ matrix.host }}
+      - name: Build host ${{ matrix.host.name }}
         # continue-on-error: true
-        run: nix develop .#gh-actions --command nixos-rebuild --flake .#${{ matrix.host }} --log-format bar-with-logs build
+        run: nix develop .#gh-actions --command nixos-rebuild --flake .#${{ matrix.host.name }} --log-format bar-with-logs --show-trace build
       - name: Debug with tmate (failure only)
         if: failure()
         uses: mxschmitt/action-tmate@v3
+        timeout-minutes: 10

--- a/flake.nix
+++ b/flake.nix
@@ -42,7 +42,7 @@
       inputs.nixpkgs.follows = "nixpkgs";
     };
     smartp = {
-      url = "github:Gigahawk/smartp";
+      url = "github:Gigahawk/smartp/migrate-to-uv";
       inputs.nixpkgs.follows = "nixpkgs";
     };
     inventree = {
@@ -160,6 +160,17 @@
       images = {
         haro = mkSdImage "haro";
       };
+      githubActions.matrix = builtins.map (name: rec {
+        inherit name;
+        inherit (self.nixosConfigurations.${name}.pkgs.stdenv.hostPlatform) system;
+        runner =
+          if system == "x86_64-linux" then
+            "ubuntu-latest"
+          else if system == "aarch64-linux" then
+            "ubuntu-24.04-arm"
+          else
+            "UNSUPPORTED-${system}";
+      }) (builtins.attrNames self.nixosConfigurations);
       nixosConfigurations = {
         # Main server
         ptolemy = lib.nixosSystem {

--- a/hosts/arios/configuration.nix
+++ b/hosts/arios/configuration.nix
@@ -53,9 +53,9 @@
     "aarch64-linux"
   ];
 
-  systemd.sleep.extraConfig = ''
-    HibernateMode=shutdown
-  '';
+  systemd.sleep.settings.Sleep = {
+    HibernateMode = "shutdown";
+  };
 
   age.secrets = {
     jasper = {

--- a/hosts/arios/hw-config.nix
+++ b/hosts/arios/hw-config.nix
@@ -48,7 +48,7 @@
   };
 
   boot.initrd.luks = {
-    reusePassphrases = true;
+    # reusePassphrases = true;
     devices = {
       nixos_decrypted = {
         device = "/dev/disk/by-label/nixos_encrypted";

--- a/hosts/haro/hw-config.nix
+++ b/hosts/haro/hw-config.nix
@@ -7,6 +7,11 @@
   ...
 }:
 {
+  boot.kernel.sysctl = {
+    # Workaround for
+    # https://github.com/NixOS/nixpkgs/issues/513512
+    "vm.mmap_rnd_bits" = 24;
+  };
 
   raspi4.config = options.raspi4.config.default // {
     cm4 = {
@@ -28,7 +33,7 @@
 
   hardware.deviceTree = {
     enable = true;
-    filter = "bcm2711-rpi-cm4.dtb";
+    filter = lib.mkForce "bcm2711-rpi-cm4.dtb";
     overlays = [
       rec {
         name = "dwc2";

--- a/hosts/ptolemy/configuration.nix
+++ b/hosts/ptolemy/configuration.nix
@@ -275,7 +275,6 @@
   services.samba-wsdd.enable = true;
   services.samba = {
     enable = true;
-    package = pkgs.sambaFull;
     openFirewall = true;
     settings = {
       global = {

--- a/hosts/ptolemy/hw-config.nix
+++ b/hosts/ptolemy/hw-config.nix
@@ -59,7 +59,7 @@
   };
 
   boot.initrd.luks = {
-    reusePassphrases = true;
+    # reusePassphrases = true;
     devices = {
       nixos_decrypted = {
         device = "/dev/disk/by-label/nixos_encrypted";

--- a/hosts/virtualbox/hw-config.nix
+++ b/hosts/virtualbox/hw-config.nix
@@ -33,7 +33,7 @@
   };
 
   boot.initrd.luks = {
-    reusePassphrases = true;
+    # reusePassphrases = true;
     devices = {
       nixos_decrypted = {
         device = "/dev/disk/by-label/nixos_encrypted";

--- a/modules/neovim/nvim-config.nix
+++ b/modules/neovim/nvim-config.nix
@@ -142,7 +142,7 @@
       sql.enable = true;
       java.enable = true;
       kotlin.enable = true;
-      ts.enable = true;
+      typescript.enable = true;
       go.enable = true;
       lua.enable = true;
       zig.enable = true;
@@ -179,14 +179,14 @@
     statusline = {
       lualine = {
         enable = true;
-        theme = "catppuccin";
+        theme = "dracula";
       };
     };
 
     theme = {
       enable = true;
-      name = "catppuccin";
-      style = "mocha";
+      name = "dracula";
+      style = "dark";
       transparent = false;
     };
 

--- a/modules/samba-users/module.nix
+++ b/modules/samba-users/module.nix
@@ -9,6 +9,7 @@ let
   cfg = config.services.samba-users;
   usersFile = pkgs.writeText "users.json" (builtins.toJSON cfg.users);
   settingsFormat = pkgs.formats.json { };
+  samba = pkgs.samba4Full;
 in
 {
   options.services.samba-users = {
@@ -66,14 +67,14 @@ in
         ExecStartPre = "+${pkgs.writers.writeBash "samba-users" ''
           echo "Deleting existing samba users"
 
-          old_users=$(${pkgs.samba4Full}/bin/pdbedit -L)
+          old_users=$(${samba}/bin/pdbedit -L)
           while IFS= read -r line; do
             username=$(echo "$line" | cut -d: -f1)
             if [[ -z "$username" ]]; then
               continue
             fi
             echo "Deleting samba user $username"
-            ${pkgs.samba4Full}/bin/smbpasswd -x "$username"
+            ${samba}/bin/smbpasswd -x "$username"
           done <<<"$old_users"
 
 
@@ -90,7 +91,7 @@ in
               continue
             fi
             echo "Creating user $username from password file $password_file"
-            echo "$password_input" | ${pkgs.samba4Full}/bin/smbpasswd -s -a "$username"
+            echo "$password_input" | ${samba}/bin/smbpasswd -s -a "$username"
           done <<<"$new_users"
         ''}";
 

--- a/users/jasper/home.nix
+++ b/users/jasper/home.nix
@@ -189,7 +189,7 @@ in
         fi
       '';
       passmenu = pkgs.writeShellScript "passmenu.sh" ''
-        selected=$(gopass ls --flat | walker --dmenu)
+        selected=$(gopass ls --flat | fuzzel --dmenu)
 
         if [[ -z "$selected" ]]; then
           exit 0
@@ -268,7 +268,7 @@ in
       settings = {
         "$mod" = "SUPER";
         "$term" = "ghostty";
-        "$menu" = "walker";
+        "$menu" = "fuzzel";
         "$bar" = "waybar";
 
         input = {
@@ -359,20 +359,19 @@ in
       };
     };
 
-  services.walker = {
+  programs.fuzzel = {
     enable = desktop;
     settings = {
-      app_launch_prefix = "";
-      as_window = false;
-      close_when_open = false;
-      disable_click_to_close = false;
-      force_keyboard_focus = false;
-      hotreload_theme = false;
-      locale = " ";
-      monitor = "";
-      terminal_title_flag = "";
-      theme = "default";
-      timeout = 0;
+      # https://github.com/kuripa/oxocarbon-fuzzel
+      colors = {
+        background = "161616ff";
+        text = "ffffffff";
+        match = "ee5396ff";
+        selection-match = "ee5396ff";
+        selection = "262626ff";
+        selection-text = "33b1ffff";
+        border = "525252ff";
+      };
     };
   };
 

